python/spinn/fat_stack.py

```diff
@@ -572,9 +572,9 @@ def run_spinn(self, example, train, use_internal_parser,
 
     def run_mlp(self, h, train):
         # Pass through MLP Classifier.
-        if self.use_difference_feature or self.use_product_feature:
-            batch_size, h_dim = h.shape[:2]
-            prem, hyp = h[:, :h_dim/2], h[:, h_dim/2:]
+        batch_size, h_dim = h.shape[:2]
+        prem, hyp = h[:, :h_dim / 4], h[:, h_dim / 2:3 * h_dim / 4]
+        h = F.concat([prem, hyp], axis=1)  # Strip off 'c' states.   
```

Comment:
Might be cleaner to have this in a conditional so that it works for the single sentence configuration. Something like:

```
        batch_size = h.shape[0]
        if self.use_sentence_pair:
            prem, hyp = h[:, :self.stack_hidden_dim], h[:, stack_hidden_dim*2:stack_hidden_dim*3]
            h = F.concat([prem, hyp], axis=1)
            ...
        else:
            h = h[:, :self.stack_hidden_dim]
```

Might need to set `self.use_sentence_pair` in the init method.